### PR TITLE
Feature/add book comments

### PR DESCRIPTION
**Pull request by w-agustin-vazquez**
## Summary

Add book comments into book detail

#### Galaxy S (or similar)

https://user-images.githubusercontent.com/77621131/147601493-0c21df63-fcff-4d7e-a882-673294cfd02a.mov

#### Pixel 2/3 (or similar)
image

#### Pixel 3XL (or similar)
image

#### iPhone SE (or similar)

https://user-images.githubusercontent.com/77621131/147602155-bfc9f59a-e7ea-4a8e-b3cb-4db526ad516e.mov

#### iPhone 6/7/8 (or similar)
image

#### iPhone 11/12 Base/Pro/Pro Max (or similar)
image

## Trello Card

[Tabla de comentarios](https://trello.com/c/6RFbFgpT)

### CODE REVIEW

File: wbooks/src/app/components/FlatListWrapper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Taked from: https://github.com/facebook/react-native/issues/31697#issuecomment-860447695
+// Solve "VirtualizedLists should never be nested inside plain ScrollViews with the same orientation"
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import React from 'react';
+import { FlatListProps, FlatList as NativeFlatList, View } from 'react-native';
+
+// Custom FlatList that doesnt try to virtualize when scrollEnabled={false}
+// Do this to avoid "VirtualizedLists should never be nested inside plain ScrollViews with the same orientation" error in console
```
Comment (guidoprinc) [inline wbooks/src/app/components/FlatListWrapper/index.tsx:9]: remove comments

File: wbooks/src/app/components/FlatListWrapper/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+export default function FlatListWrapper<ItemT>(props: FlatListProps<ItemT>) {
```
Comment (guidoprinc) [inline wbooks/src/app/components/FlatListWrapper/index.tsx:10]: what is the utility of this component?

File: wbooks/src/app/screens/Library/screens/BookDetail/components/Comment/styles.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { StyleSheet } from 'react-native';
+
+export const styles = StyleSheet.create({
+  comment: {
+    flexDirection: 'row',
+    paddingTop: 15
+  },
+  image: {
+    width: 45,
+    height: 45,
```
Comment (guidoprinc) [inline wbooks/src/app/screens/Library/screens/BookDetail/components/Comment/styles.ts]: extract same square size value into a constant variable

File: wbooks/src/app/screens/Library/screens/BookDetail/components/Comments/components/CommentsFooter/index.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+interface Props {
+  handleViewAll: () => void;
+  hasMoreComments?: boolean;
+}
+
+const CommentsFooter = ({ handleViewAll, hasMoreComments }: Props) => {
+  if (!hasMoreComments) return null;
+
+  return (
+    <Pressable onPress={handleViewAll}>
```
Comment (guidoprinc) [inline wbooks/src/app/screens/Library/screens/BookDetail/components/Comments/components/CommentsFooter/index.tsx]: do not use Pressable, use TouchableOpacity or TouchableWithoutFeedback

**Review by guidoprinc — CHANGES_REQUESTED** [review]

File: wbooks/src/app/components/FlatListWrapper/index.tsx (hunk 2 above)
Comment (w-agustin-vazquez) [reply wbooks/src/app/components/FlatListWrapper/index.tsx:10]: I take it from [here](https://github.com/facebook/react-native/issues/31697#issuecomment-860447695), the purpose it's remove the warning "VirtualizedLists should never be nested inside plain ScrollViews with the same orientation" when use a ScrollView and a FlatList. Without the ScrollView the view isnt be scrolleable.

File: wbooks/src/app/components/Layout/index.tsx [hunk 5]
```diff
@@ -1,10 +1,15 @@
 import React, { PropsWithChildren } from 'react';
 import { View } from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
```
Comment (guidoprinc) [inline wbooks/src/app/components/Layout/index.tsx]: use ScrollView from react-native

File: wbooks/src/app/screens/Library/screens/BookDetail/components/Comments/components/CommentsFooter/styles.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { AppColors } from '@constants/colors';
+import { StyleSheet } from 'react-native';
```
Comment (guidoprinc) [inline wbooks/src/app/screens/Library/screens/BookDetail/components/Comments/components/CommentsFooter/styles.ts:1]: move this line above. Remember that library imports goes first

**Review by guidoprinc — CHANGES_REQUESTED** [review]

File: wbooks/src/app/components/FlatListWrapper/index.tsx (hunk 2 above)
Comment (w-agustin-vazquez) [reply wbooks/src/app/components/FlatListWrapper/index.tsx:10]: Should i remove it? Exists another solution?👀

Comment (guidoprinc) [reply wbooks/src/app/components/FlatListWrapper/index.tsx:10]: there are other solutions, but this is ok for the training

**Review by guidoprinc — APPROVED** [review]

**Pull request merged**